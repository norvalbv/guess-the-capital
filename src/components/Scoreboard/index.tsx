import React, { ReactElement } from 'react';

const Scoreboard = (): ReactElement => {
  return <div>Scoreboard</div>;
};

export default Scoreboard;
