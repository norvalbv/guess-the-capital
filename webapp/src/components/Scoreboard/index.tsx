import React, { ReactElement } from 'react';

type ScoreboardProps = {
  score: { win: number; lose: number };
};

const Scoreboard = ({ score }: ScoreboardProps): ReactElement => {
  return (
    <section className="text-sm text-white">
      Win: <span className="text-green-300">{score.win}</span> | Lose:{' '}
      <span className="text-red-300">{score.lose}</span>
    </section>
  );
};

export default Scoreboard;
