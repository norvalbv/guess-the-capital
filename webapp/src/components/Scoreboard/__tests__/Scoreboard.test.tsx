import React from 'react';
import renderer from 'react-test-renderer';

import Scoreboard from '..';

describe('<Scoreboard />', () => {
  test('Should render with mandatory props.', () => {
    const tree = renderer.create(<Scoreboard score={{ win: 1, lose: 1 }} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
