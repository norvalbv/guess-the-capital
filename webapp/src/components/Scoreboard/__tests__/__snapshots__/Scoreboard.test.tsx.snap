// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Scoreboard /> Should render with mandatory props. 1`] = `
<section
  className="text-sm text-white"
>
  Win: 
  <span
    className="text-green-300"
  >
    1
  </span>
   | Lose:
   
  <span
    className="text-red-300"
  >
    1
  </span>
</section>
`;
